dvc/fs/git.py

```diff
@@ -1,52 +1,88 @@
 import errno
 import os
+from typing import (
+    TYPE_CHECKING,
+    Any,
+    BinaryIO,
+    Callable,
+    Dict,
+    Optional,
+    Tuple,
+)
 
-from tqdm.utils import CallbackIOWrapper
+from fsspec.spec import AbstractFileSystem
 
-from dvc.utils import is_exec, relpath
+if TYPE_CHECKING:
+    from io import BytesIO
 
-from ..progress import DEFAULT_CALLBACK
-from .base import FileSystem
+    from dvc.scm.git import Git
+    from dvc.scm.git.objects import GitTrie
 
 
-class GitFileSystem(FileSystem):  # pylint:disable=abstract-method
-    """Proxies the repo file access methods to Git objects"""
+def bytesio_len(obj: "BytesIO") -> Optional[int]:
+    try:
+        offset = obj.tell()
+        length = obj.seek(0, os.SEEK_END)
+        obj.seek(offset)
+    except (AttributeError, OSError):
+        return None
+    return length
 
-    sep = os.sep
 
-    scheme = "local"
+class GitFileSystem(AbstractFileSystem):
```

Comment:
This implementation is not complete but is compatible with the previous implementation. We may need to iterate separately on `scmrepo`.